python/python/tests/test_scan.py

```diff
@@ -1,16 +1,100 @@
-from __future__ import annotations
-
-from typing import List, cast
+import asyncio
+import re
+from typing import AsyncGenerator, List, cast
 
 import pytest
-from glide import ClusterScanCursor
+from glide import ByAddressRoute
 from glide.async_commands.command_args import ObjectType
 from glide.config import ProtocolVersion
 from glide.exceptions import RequestError
+from glide.glide import ClusterScanCursor
 from glide.glide_client import GlideClient, GlideClusterClient
+from tests.conftest import create_client
+from tests.utils.cluster import ValkeyCluster
 from tests.utils.utils import get_random_string
 
 
+# Helper function to get a number of nodes, and ask the cluster till we get the number of nodes
+async def is_cluster_ready(client: GlideClusterClient, count: int) -> bool:
```

Comment:
Why make this function async, when the only caller is just waiting for its completion?

Reply:
Not sure i understand the question?
It uses async calls, so it has to be async, I can't await in sync function.

Reply:
Ah, I see. Didn't realize this is part of the python language restriction. Sorry about that.